packages/central-server/src/apiV2/userAccounts/EditUserAccounts.js

```diff
@@ -37,13 +39,39 @@ export class EditUserAccounts extends EditHandler {
     // Update Record
     const { password, profile_image: profileImage, ...restOfUpdatedFields } = this.updatedFields;
     let updatedFields = restOfUpdatedFields;
+
     if (password) {
       updatedFields = {
         ...updatedFields,
         ...hashAndSaltPassword(password),
       };
     }
 
+    // Check if there are any updated user preferences in the request
+    const updatedUserPreferences = Object.entries(updatedFields).filter(([key]) =>
+      USER_PREFERENCES_FIELDS.includes(key),
+    );
+
+    // If there are, extract them and save them with the existing user preferences
+    if (updatedUserPreferences) {
+      // Remove user preferences fields from updatedFields so they don't get saved else where
+      updatedFields = Object.keys(updatedFields).filter(
+        field => !USER_PREFERENCES_FIELDS.includes(field),
+      );
+
+      const userRecord = await this.models.user.findById(this.recordId);
+      const { preferences } = userRecord;
+
+      const updatedPreferenceFields = updatedUserPreferences.reduce((obj, [key, value]) => {
+        return { ...obj, [key]: value };
+      }, {});
+
+      updatedFields = {
+        preferences: { ...preferences, ...updatedPreferenceFields },
+        ...updatedFields,
+      };
```

Comment:
One note here. If someone sends a `preferences` field it will override all of this.

```
updatedFields = {
  preferences: { country_id: 'DL' },
  project_id: 'explore',
}
```
Something like this would only submit the `country_id`, do we want to combine with the passed in `preferences` field? Or maybe we should error if someone tries to send a `preferences` field? 🤔

Reply:
Good catch thanks. I think that the intention of my api is that it only allows setting specified fields (for now anyway) so I like the idea of throwing an error if a preferences field is sent.